src/modules/lua/debug_lua.c

```diff
@@ -168,61 +206,77 @@ static void ldbList(int around, int context) {
 }
 
 /* Append a human readable representation of the Lua value at position 'idx'
- * on the stack of the 'lua' state, to the SDS string passed as argument.
- * The new SDS string with the represented value attached is returned.
+ * on the stack of the 'lua' state, to the string passed as argument.
+ * The new string with the represented value attached is returned.
  * Used in order to implement ldbLogStackValue().
  *
  * The element is neither automatically removed from the stack, nor
  * converted to a different type. */
 #define LDB_MAX_VALUES_DEPTH (LUA_MINSTACK / 2)
-static sds ldbCatStackValueRec(sds s, lua_State *lua, int idx, int level) {
+static ValkeyModuleString *ldbCatStackValueRec(ValkeyModuleString *s, lua_State *lua, int idx, int level) {
     int t = lua_type(lua, idx);
+    ValkeyModuleString *old_s = NULL;
+    const char *prefix = ValkeyModule_StringPtrLen(s, NULL);
 
-    if (level++ == LDB_MAX_VALUES_DEPTH) return sdscat(s, "<max recursion level reached! Nested table?>");
+    if (level++ == LDB_MAX_VALUES_DEPTH) {
+        const char *msg = "<max recursion level reached! Nested table?>";
+        ValkeyModule_StringAppendBuffer(NULL, s, msg, strlen(msg));
+        return s;
+    }
 
     switch (t) {
     case LUA_TSTRING: {
         size_t strl;
         char *strp = (char *)lua_tolstring(lua, idx, &strl);
-        s = sdscatrepr(s, strp, strl);
+        ValkeyModule_StringAppendBuffer(NULL, s, strp, strl);
     } break;
-    case LUA_TBOOLEAN: s = sdscat(s, lua_toboolean(lua, idx) ? "true" : "false"); break;
-    case LUA_TNUMBER: s = sdscatprintf(s, "%g", (double)lua_tonumber(lua, idx)); break;
-    case LUA_TNIL: s = sdscatlen(s, "nil", 3); break;
+    case LUA_TBOOLEAN: {
+        const char *bool_str = lua_toboolean(lua, idx) ? "true" : "false";
+        ValkeyModule_StringAppendBuffer(NULL, s, bool_str, strlen(bool_str));
+        break;
+    }
+    case LUA_TNUMBER: {
+        old_s = s;
+        s = ValkeyModule_CreateStringPrintf(NULL, "%s %g", prefix, (double)lua_tonumber(lua, idx));
```

Comment:
In the TNUMBER case, isn't there an extra space added in the format `"%s %g"`? Before this change, it used sdscatprintf with the format `"%g"` which doesn't insert a space before the number.

I see `old_s` is freed in the end of the function. I think the code is easier to follow if make this variable local to this case and free it locally. Same with the `prefix` variable, either move it to the local scope or we don't need it at all?

```suggestion
        ValkeyModuleString old_s = s;
        s = ValkeyModule_CreateStringPrintf(NULL, "%s%g",
                                            ValkeyModule_StringPtrLen(s, NULL),
                                            (double)lua_tonumber(lua, idx));
        ValkeyModule_FreeString(NULL, old_s);
```